### DISCUSSION TITLE
Adding SLES support to sap_ha_prepare_pacemaker role

### DISCUSSION THREAD

**Discussion by mr-stringer** [sap-linuxlab/community.sap_install#232] [General]
I've started to look at adding SLES support to the `sap_ha_prepare_pacemaker` role and then do the same with other `sap_ha_*` roles.  Currently, the code is almost all RedHat specific so requires the ability to branch on OS type (probably through the `ansible_distribution` fact).

My question is should I repeat the pattern found in roles like `sap_hana_preconfigure` where, based on OS, the code sources OS specific task files?  This seems likes the cleanest approach and will help reduce the amount of Ansible output.  But I want to get a little feedback before I start writing this.

Cheers, Steve.

**Comment by richm** [comment]
See also https://github.com/linux-system-roles/ha_cluster/issues/122

**Comment by ja9fuchs** [comment]
The former role `sap_ha_prepare_pacemaker` does not exist anymore. This code is not compatible with the replacement role, `sap_ha_pacemaker_cluster`, which wraps around the `ha_cluster` Linux System Role to avoid duplication and increase flexibility for users. 

Please see richm's comment above - the issue to support crmsh for other OS distributions has been raised with the `ha_cluster` role.

**Comment by Wabri** [comment]
@marcelmamula is this something that you are working on?

**Comment by marcelmamula** [comment]
SLES15 and SLES4SAP 15 support was added to ha_cluster underlying role in https://github.com/linux-system-roles/ha_cluster/pull/186 and sap_ha PR https://github.com/sap-linuxlab/community.sap_install/pull/642 is under review now. This issue can be closed.
